src/middlewared/middlewared/plugins/acme_protocol_/client_utils.py

```diff
@@ -0,0 +1,50 @@
+import json
+
+import josepy as jose
+from acme import client, messages
+
+
+def get_acme_client_and_key(data: dict) -> tuple[client.ClientV2, jose.JWKRSA]:
+    """
+    Expected data dict should contain the following
```

Comment:
If interested, you can use `typing.TypedDict` to describe this schema more formally. It then can be re-used, IDE autocompletion will work better